core/templates/dev/head/pages/exploration_editor/feedback_tab/ThreadDataService.js

```diff
@@ -46,16 +52,53 @@ oppia.factory('ThreadDataService', [
 
     var _fetchThreads = function(successCallback) {
       var fPromise = $http.get(_THREAD_LIST_HANDLER_URL);
-      var sPromise = $http.get(_SUGGESTION_LIST_HANDLER_URL, {
-        params: {
-          list_type: 'all',
-          has_suggestion: true
-        }
-      });
+      var sPromise;
+      if (constants.USE_NEW_SUGGESTION_FRAMEWORK) {
+        sPromise = $http.get(_SUGGESTION_LIST_HANDLER_URL, {
+          params: {
+            list_type: 'target',
+            target_type: 'exploration',
+            target_id: _expId
+          }
+        });
+      } else {
+        sPromise = $http.get(_SUGGESTION_LIST_HANDLER_URL, {
+          params: {
+            list_type: 'all',
+            has_suggestion: true
+          }
+        });
+      }
 
       $q.all([fPromise, sPromise]).then(function(res) {
         _data.feedbackThreads = res[0].data.threads;
-        _data.suggestionThreads = res[1].data.threads;
+        _data.suggestionThreads = [];
+        if (constants.USE_NEW_SUGGESTION_FRAMEWORK) {
+          for (var i = 0; i < res[1].data.suggestions.length; i++) {
+            suggestion = res[1].data.suggestions[i];
+            suggestion.thread_id = suggestion.suggestion_id.slice(
+              suggestion.suggestion_id.indexOf('.') + 1);
+            _data.suggestionThreads.push(suggestion);
+          }
+          for (var i = 0; i < _data.suggestionThreads.length; i++) {
+            for (var j = 0; j < _data.feedbackThreads.length; j++) {
+              if (_data.suggestionThreads[i].thread_id ===
+                _data.feedbackThreads[j].thread_id) {
```

Comment:
Suggest indenting this line 2 spaces to separate if-statement from the code inside.

Reply:
Done